cmd/tempo/app/app.go

```diff
@@ -231,6 +240,14 @@ func (t *App) Run() error {
 	handler := signals.NewHandler(t.Server.Log())
 	go func() {
 		handler.Loop()
+
+		shutdownRequested.Store(true)
+		t.Server.SetKeepAlivesEnabled(false)
+
+		if t.cfg.ShutdownDelay > 0 {
+			time.Sleep(t.cfg.ShutdownDelay)
```

Comment:
Do we also want to sleep when there are no requests in flight?

Reply:
i'm concerned about the complexity about that change. this is a very simple one that will do nearly all of what we want with some config choices in k8s